doc/neps/ufunc-overrides.rst

```diff
@@ -575,14 +591,78 @@ For option (2), the relevant example is the fourth, with ``arr *=
+is viral, in the sense that any other class that wishes to support binary
+operations with your class now must also follow these rules for supporting
+binary arithmetic with :class:`ndarray` (i.e., they must either implement
+``__array_ufunc__`` or set it to :obj:`None`). We believe this is a good
+thing, because it ensures the consistency of ufuncs and arithmetic on all
+objects that support them.
+
+To make implementing such array-like classes easier, the mixin class
+:class:`~numpy.lib.mixins.NDArrayOperatorsMixin` provides option (1) style
+overrides for all binary operators with corresponding Ufuncs. Classes
+that wish to implement ``__array_ufunc__`` for compatible versions
+of NumPy but that also need to support binary arithmetic with NumPy arrays
+on older versions, should ensure that ``__array_ufunc__`` can also be used
+to implement all binary operations they support.
+
 Finally, we note that we had extensive discussion about whether it might
 make more sense to ask classes like ``MyObject`` to implement a full
 ``__array_ufunc__`` [6]_. In the end, allowing classes to opt out was
 preferred, and the above reasoning led us to agree on a similar
-implementation for :class:`ndarray` itself. To help implement array-like
-classes, the mixin class :class:`~numpy.lib.mixins.NDArrayOperatorsMixin`
-provides overrides for all binary operators with corresponding ufuncs.
-
+implementation for :class:`ndarray` itself. We decided to require disabling
+ufuncs to opt out to ensure that a class cannot define Ufuncs to return
+different results than the corresponding binary operations (i.e., if
+``np.add(x, y)`` is defined, it should match ``x + y``).
+
+.. [9] http://bugs.python.org/issue30140
+
+.. _neps.ufunc-overrides.list-operations:
+
+List of binary operations and NumPy Ufuncs
+------------------------------------------
+
+Here is a full list of Python binary operations and the corresponding NumPy
+Ufuncs used by :class:`ndarray`:
+
+====== ============ =========================================
+Symbol Operator     NumPy Ufunc(s)
+====== ============ =========================================
+``<``  ``lt``       :func:`less`
+``<=`` ``le``       :func:`less_equal`
+``==`` ``eq``       :func:`equal`
+``!=`` ``ne``       :func:`not_equal`
+``>``  ``gt``       :func:`greater`
+``>=`` ``ge``       :func:`greater_equal`
+``+``  ``add``      :func:`add`
+``-``  ``sub``      :func:`subtract`
+``*``  ``mul``      :func:`multiply`
+``/``  ``truediv``  :func:`true_divide`
+       (Python 3)
+``/``  ``div``      :func:`divide`
+       (Python 2)
+``//`` ``floordiv`` :func:`floor_divide`
+``%``  ``mod``      :func:`mod`
+``**`` ``pow``      :func:`power`
+``<<`` ``lshift``   :func:`left_shift`
+``>>`` ``rshift``   :func:`right_shift`
+``&``  ``and_``     :func:`bitwise_and`
+``^``  ``xor_``     :func:`bitwise_xor`
+``|``  ``or_``      :func:`bitwise_or`
+\      ``divmod``   :func:`floor_divide`, :func:`mod` [*]_
```

Comment:
I put in "NA" instead. I am concerned that "--" could be confused with an infix operator.